jerry-core/ecma/operations/ecma-objects-general.c

```diff
@@ -79,12 +79,22 @@ ecma_op_create_object_object_arg (ecma_value_t value) /**< argument of construct
 {
   ecma_check_value_type_is_spec_defined (value);
 
-  if (ecma_is_value_object (value)
-      || ecma_is_value_number (value)
-      || ecma_is_value_prop_name (value)
-      || ecma_is_value_boolean (value))
+  if (!ecma_is_value_undefined (value)
+      && !ecma_is_value_null (value))
   {
     /* 1.b, 1.c, 1.d */
+#if ENABLED (JERRY_ESNEXT)
+    JERRY_ASSERT (ecma_is_value_object (value)
+                  || ecma_is_value_number (value)
+                  || ecma_is_value_prop_name (value)
+                  || ecma_is_value_boolean (value)
+                  || ecma_is_value_bigint (value));
+#else
```

Comment:
Please use this macro:
https://github.com/jerryscript-project/jerryscript/blob/master/jerry-core/ecma/base/ecma-helpers.h#L211

The `#else` has no comment, but it should be removed anyway.